lib.php

```diff
@@ -45,7 +45,7 @@ function availability_proctor_after_require_login() {
 
     // User is trying to start an attempt, redirect to proctor if it is not started.
     $scriptname = isset($_SERVER['SCRIPT_NAME']) ? $_SERVER['SCRIPT_NAME'] : null;
-    if ($scriptname == '/mod/quiz/startattempt.php') {
+   if (($scriptname == '/mod/quiz/startattempt.php') || (strpos($_SERVER['SCRIPT_NAME'], '/mod/quiz/startattempt.php') !== false)) 
```

Comment:
Should be
```php
$scriptname = isset($_SERVER['SCRIPT_NAME']) ? $_SERVER['SCRIPT_NAME'] : '';
if (strpos($scriptname , '/mod/quiz/startattempt.php') !== false) {
```
Note that unset SCRIPT_NAME should return string, and curly brace that you've missed